src/cli/error.rs

```diff
@@ -0,0 +1,42 @@
+use std::{ffi::OsString, fmt};
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug)]
+pub enum Error {
+    DisplayHelp,
+    DisplayVersion,
+    LexOpt(lexopt::Error),
+    UnexpectedArg(String),
+    ConflictingFlags(&'static [&'static str]),
+    MissingRequires(&'static str, &'static str),
+    FlagMissingValue(&'static str),
+    InvalidFlagValue(&'static str, OsString),
+}
+
+impl From<lexopt::Error> for Error {
+    fn from(e: lexopt::Error) -> Self {
+        Self::LexOpt(e)
+    }
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::DisplayHelp => f.write_str("Signals to display help"),
+            Self::DisplayVersion => f.write_str("Signals to display version"),
```

Comment:
Those cases aren't meant to happen, so I would suggest moving `DisplayHelp` and `DisplayVersion` into another enum and change `try_parse_args`' signature:

```rust
enum ParseResult {
    Ok(Args),
    DisplayHelp,
    DisplayVersion,
    Err(Error),
}
```

Since we're not doing anything fancy like `?`, I wouldn't mind the `Err(_)` variant be inside `ParseResult` instead of wrapping it into a `Result`.